File: custom_services/src/service_server.cpp
#include "ros/ros.h"
#include "custom_services/MeanThreeInts.h"

bool mean(custom_services::MeanThreeInts::Request  &req,
          custom_services::MeanThreeInts::Response &res)
{
  res.mean = (req.a + req.b + req.c)/3.0;
  ROS_INFO("request: x=%ld, y=%ld, z=%ld", (long int)req.a, (long int)req.b, (long int)req.c);
  ROS_INFO("sending back response: [%f]", res.mean);
  return true;
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "service_server");
  ros::NodeHandle n;

  ros::ServiceServer service = n.advertiseService("mean_three_ints", mean);
  ROS_INFO("Ready to calculate mean.");
  ros::spin();

  return 0;
}

File: custom_services/scripts/service_server.py
#!/usr/bin/env python

from custom_services.srv import *
import rospy

def handle_mean_three_ints(req: MeanThreeIntsRequest):
    print("Returning [(%s+%s+%s)/3 = %f]"%(req.a, req.b, req.c, (req.a + req.b +req.c)/3))
    return MeanThreeIntsResponse((req.a + req.b +req.c)/3)

def service_server():
    rospy.init_node('service_server')
    s = rospy.Service('mean_three_ints', MeanThreeInts, handle_mean_three_ints)
    print("Ready to calculate mean of three ints.")
    rospy.spin()

if __name__ == "__main__":
    service_server()
